packages/data-models/appConfig.ts

```diff
@@ -51,6 +51,15 @@ const APP_LANGUAGES = {
   default: "gb_en",
 };
 
+export interface ILanguageMeta {
+  rtl?: boolean;
+}
+const APP_LANGUAGES_META: { [languageCode: string]: ILanguageMeta } = {
```

Comment:
nit(non-blocking) - Not sure we really need the `APP_` prefix, in my mind that is more for cases of app-specific UI, not so much underlying settings (although not remotely consistent right now). But given this all sits in `appConfig` I feel like something along the lines of `appConfig.LANGUAGE_SETTINGS` or `appConfig.LANGUAGE_CONFIG` or `appConfig.LANGUAGES` might be a little more human readable than `appConfig.APP_LANGUAGES_META`... but I don't have an overly strong opinion (open to your thoughts too @esmeetewinkel )

Reply:
Not sure I feel strongly either, but do also like to keep things human readable if possible. 

We do have an existing property `config.app_config.APP_LANGUAGES.default` which should maybe also be taken into account. I think currently the full list of languages is not specified in the config, is that correct? So you would only need to specify a language in the config if you want to change any of its default settings? 

Maybe something like: `config.app_config.LANGUAGES = { en: { default: true} , en_rtl: { rtl: true } }`

(default values of `default` and `rtl` are both `false`)

Reply:
> I think currently the full list of languages is not specified in the config, is that correct?
Yes that's right. I think that there is no master list of available languages currently and any value can be passed as the language code to the `emit: set_language` action. So a deployment's available languages are just those that are authored to be passed as values to that action, from any template, and it's up to the authors to ensure that the specified language codes have associated translations and assets. This seems a bit fragile, but a full overhaul is maybe outside of the scope of this PR. Requiring authors to specify the available languages in the deployment config would make sense to me, then when setting the language via the `emit: set_language` action we could run a check against the specified languages, throwing an error if the target language isn't in the list (this is what we do for the skins and themes, for example).

> Maybe something like: `config.app_config.LANGUAGES = { es: { default: true } , en_rtl: { rtl: true } }`
I like this suggestion, you're right that having separate system additional to `config.app_config.APP_LANGUAGES.default` doesn't make much sense. One negative I can see is that this syntax requires authors (at least those who are editing the deployment config) to explicitly write and understand nested javascript objects. But we already do require this (for example for the Firebase config settings) so I don't think it's a problem.

So a few of questions probably for @chrismclarke:
1. If we want to migrate to a system where languages and their metadata are handled within a single property in the app config, can we do better than the proposed syntax, `config.app_config.LANGUAGES = { es: { default: true } , en_rtl: { rtl: true } }`?
  - In this syntax, adding a language without any non-default metadata would require adding an empty object, e.g. `en_gb: {}` or else arbitrarily specifying a default property, e.g. `en_gb: { default: false }`. For themes and skins, we use an array to list the available entities and then use a separate property to specify which one is default. But here we already have two metadata properties (`default` and `rtl`) and may want to expand to have more in the future.
2. How would be best to handle the migration to this new system to minimise breaking changes to all deployment configs?
  - I'm thinking it would be sensible to mark the `APP_LANGUAGES` property as deprecated but continue to support it. This property is only used to get the name of the default language, so it should be straightforward to look in the new `LANGUAGES` property for a default language instead (with a fallback to the deprecated property)
3. Would it be desirable to require an exhaustive list of available languages in the config, and throw an error(/warning?) when trying to set the language to one that is not explicitly named?

Reply:
Good questions - my initial thoughts would probably be:

1. Probably better to store as an array so we can use the order to imply default if not explicitly set (or make it always first entry), and track more metadata from the start, e.g.
```
[
{country_code:'tz', language_code:'sw', label: 'Swahili'},
{country_code:'za', language_code:'af', label: 'Afrikaans'},
]
```
This would then allow authors in the future to more easily filter languages by country if useful to do so. If wanting to pass from a sheet then probably we could include a `fromSheet(...)` operator in the config that takes the name and reads in from there (again, future planning). At least if we have the list in the config then we can run QC checks to make sure language exists, and expose a protected variable to allow authors to see available list and select, e.g. `@field._language_options` or `@filed._language` (for active)

2. Agree using a new property and likely using scripts to migrate during deprecation process. I'd probably recommend swapping out `LANGUAGES` for `LOCALISATION` as we may use the property to contain more settings like how to handle dates within the current configuration.

3. I'd say still worth letting it be optional (if authors have hardcoded lists of languages into other sheets and want to use to select then no reason to stop them), but just use documentation to highlight benefit of using config (i.e. QC checks and protected variable population)

Reply:
Looping back to this PR then begs the question whether we want a short-term/intermediate solution or to do a slightly larger refactor now...? I don't have a strong feeling either way, I guess it depends on how urgent we want the RTL support.

Reply:
@jfmcquade - might be worth pulling out more actionable parts of this discussion to #2254